test/core/libp2p/node/ping_pong_integration_test.cpp

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <memory>
+
+#include <gtest/gtest.h>
+#include <boost/asio/io_context.hpp>
+#include <testutil/outcome.hpp>
+#include "libp2p/muxer/yamux.hpp"
+#include "libp2p/transport/all.hpp"
+#include "libp2p/host.hpp"
+#include "libp2p/host_builder.hpp"
+#include "libp2p/peer/protocol.hpp"
+
+
+using namespace libp2p;
+using namespace libp2p::transport;
+using namespace libp2p::muxer;
+using namespace libp2p::stream;
+using namespace kagome::common;
+using namespace libp2p::multi;
+using namespace libp2p::peer;
+
+/**
+ * Test, emulating two nodes exchanging ping-pong messages
+ */
+class PingPongIntegrationTest : public ::testing::Test {
+ public:
+  boost::asio::io_context context_;
+
+  const Protocol kDefaultProtocol = "/default-proto/1.0.0";
+
+  const Buffer kPingMsg = Buffer{}.put("PING");
+  const Buffer kPongMsg = Buffer{}.put("PONG");
+
+  const Multiaddress kFirstMultiaddress =
+      Multiaddress::create("/ip4/0.0.0.0/tcp/0").value();
```

Comment:
Is it ok that they are equal?

Reply:
The test is not ready yet, most likely they will be different